src/App.tsx

```diff
@@ -1,27 +1,58 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [currentPage, setCurrentPage] = useState(1);
+  const [itemsPerPage, setItemsPerPage] = useState(5);
+  const [itemsIndexOnCurrentPage, setItemsIndexOnCurrentPage] = useState(
+    getNumbers(currentPage, itemsPerPage * currentPage),
+  );
+  const total = items.length;
+  const pageCount = Math.ceil(total / itemsPerPage);
+
+  useEffect(() => {
+    setItemsIndexOnCurrentPage(
+      getNumbers(
+        itemsPerPage * (currentPage - 1),
+        itemsPerPage * currentPage > items.length
+          ? items.length - 1
+          : itemsPerPage * currentPage - 1,
+      ),
+    );
+  }, [itemsPerPage, currentPage]);
```

Comment:
As Denys said before you don't need a separate state for `itemsIndexOnCurrentPage `

General rule - if you calculate/define something based on state values or props you don't need a separate state for it. Reason - state or props change causes rerender and recalculation of common variables.
You can `useMemo` here to make this value recalculate only on related state values changes:

```suggestion
  const itemsIndexOnCurrentPage = useMemo(() => {
      return getNumbers(
        itemsPerPage * (currentPage - 1),
        itemsPerPage * currentPage > items.length
          ? items.length - 1
          : itemsPerPage * currentPage - 1,
    );
  }, [itemsPerPage, currentPage]);
```

Reply:
There shouldn't be a problem with initial values. If any - adjust `getNumbers` logic